src/Workspaces/Core/Portable/Workspace/Solution/Checksum.cs

```diff
@@ -150,7 +150,15 @@ public void WriteTo(ObjectWriter writer)
             => _checksum.WriteTo(writer);
 
         public void WriteTo(Span<byte> span)
-            => Contract.ThrowIfFalse(MemoryMarshal.TryWrite(span, ref Unsafe.AsRef(in _checksum)));
+        {
+            unsafe
+            {
+                fixed (byte* bytes = &span.GetPinnableReference())
+                {
+                    *(HashData*)bytes = _checksum;
```

Comment:
📝 This should write each field individually rather than trying to copy the entire structure to a new location.

Reply:
tried that, but i get:

```
Severity	Code	Description	File
Error	CS0191	A readonly field cannot be assigned to (except in a constructor or init-only setter of the type in which the field is defined or a variable initializer)	C:\github\roslyn\src\Workspaces\Core\Portable\Workspace\Solution\Checksum.cs
```